src/main/java/com/codeborne/selenide/impl/DownloadFileToFolderCdp.java

```diff
@@ -0,0 +1,154 @@
+      throw new IllegalStateException("Downloads folder is not configured");
+    }
+
+    folder.cleanupBeforeDownload();
+    prepareDownloadWithCdp(driver);
+
+    long downloadStartedAt = currentTimeMillis();
+
+    action.perform(driver, clickable);
+
+    waitUntilDownloadsCompleted(timeout, pollingInterval);
+
+    Downloads newDownloads = new Downloads(folder.filesNewerThan(downloadStartedAt));
+    if (log.isInfoEnabled()) {
+      log.info("Downloaded files in {}: {}", folder, newDownloads.filesAsString());
+    }
+    if (log.isDebugEnabled()) {
+      log.debug("All downloaded files: {}", folder.filesAsString());
+    }
+
+    File downloadedFile = newDownloads.firstDownloadedFile(driver, timeout, fileFilter);
+    return archiveFile(driver, downloadedFile);
+  }
+
+  @Nullable
+  protected DownloadsFolder getDownloadsFolder(Driver driver) {
+    return driver.browserDownloadsFolder();
+  }
+
+  private void waitUntilDownloadsCompleted(long timeout, long pollingInterval) {
+    for (long start = currentTimeMillis(); currentTimeMillis() - start < timeout; pause(pollingInterval)) {
+      if (downloadComplete.get()) {
+        log.info("File {} download is complete", fileName.get());
+        return;
+      }
+    }
+  }
+
+  private void pause(long milliseconds) {
+    try {
+      sleep(milliseconds);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Nonnull
+  protected File archiveFile(Driver driver, File downloadedFile) {
+    File uniqueFolder = downloader.prepareTargetFolder(driver.config());
+    File archivedFile = new File(uniqueFolder, downloadedFile.getName());
+    moveFile(downloadedFile, archivedFile);
+    log.debug("Moved the downloaded file {} to {}", downloadedFile, archivedFile);
+    return archivedFile;
+  }
+
+  private void initDevTools(Driver driver) {
+    if (driver.browser().isChromium()) {
+      devTools = ((HasDevTools) driver.getWebDriver()).getDevTools();
+      devTools.createSession();
+      devTools.send(Page.enable());
+    } else {
+        throw new RuntimeException("The browser that you choose doesn't have such functionality.");
+    }
+  }
+
+  private void prepareDownloadWithCdp(Driver driver) {
+    initDevTools(driver);
+
+    devTools.send(org.openqa.selenium.devtools.v119.browser.Browser.setDownloadBehavior(
+      org.openqa.selenium.devtools.v119.browser.Browser.SetDownloadBehaviorBehavior.ALLOW,
+      Optional.empty(),
+      Optional.of(getDownloadsFolder(driver).toString()),
+      Optional.of(true)));
+
+    devTools.addListener(org.openqa.selenium.devtools.v119.browser.Browser.downloadWillBegin(), handler -> {
+      fileName.set(handler.getSuggestedFilename());
+    });
+
+    devTools.addListener(
+      org.openqa.selenium.devtools.v119.browser.Browser.downloadProgress(),
+      e -> {
+        downloadComplete.set(e.getState() == COMPLETED);
+        log.info("Download is in progress");
```

Comment:
There appears to be a debug log message not info.

Reply:
Yes. This was a bug. I fixed it